spec/lang/representation.md

```diff
@@ -231,16 +231,58 @@ impl Type {
 
 ### Enums
 
-TODO: implement Enum decoding & encoding.
+Enum encoding and decoding.
+Note that the discriminant setting and reading does not modify any bytes.
+This is does not allow types like `Result<bool, bool>` to be stored in one byte.
+However this is fine as Rust currently does not do any optimizations like that
+and doing it here will introduce more questions.
```

Comment:
I don't understand this paragraph. How does it nit modify any bytes...?

`Result<bool, bool>` can't be stored in one byte since the types have overlapping valid byte ranges.